src/Integrations/POSCatalog/AsyncGenerator.php

```diff
@@ -0,0 +1,230 @@
+<?php
+/**
+ *  Async Generator class.
+ *
+ * @package Automattic\WooCommerce\ProductFeedForOpenAI
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\ProductFeedForOpenAI\Integrations\POSCatalog;
+
+use Automattic\WooCommerce\ProductFeedForOpenAI\Feed\ProductWalker;
+use Automattic\WooCommerce\ProductFeedForOpenAI\Feed\WalkerProgress;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Async Generator for the POS catalog.
+ */
+final class AsyncGenerator {
+	/**
+	 * The Action Scheduleraction hook for the feed generation.
+	 *
+	 * @var string
+	 */
+	const FEED_GENERATION_ACTION = 'wpfoai_pos_catalog_feed_generation';
+
+	/**
+	 * The Action Scheduler action hook for the feed deletion.
+	 *
+	 * @var string
+	 */
+	const FEED_DELETION_ACTION = 'wpfoai_pos_catalog_feed_deletion';
+
+	/**
+	 * The transient key for the feed generation status.
+	 *
+	 * @var string
+	 */
+	const TRANSIENT_KEY = 'pos_feed_status';
+
+	/**
+	 * The time limit for each batch.
+	 *
+	 * @var int
+	 */
+	const TIME_LIMIT = 5 * MINUTE_IN_SECONDS;
+
+	/**
+	 * Feed expiry time, once completed.
+	 * If the feed is not downloaded within this timeframe, a new one will need to be generated.
+	 *
+	 * @var int
+	 */
+	const FEED_EXPIRY = 20 * MINUTE_IN_SECONDS;
+
+	/**
+	 * Possible states of generation.
+	 */
+	const STATE_SCHEDULED   = 'scheduled';
+	const STATE_IN_PROGRESS = 'in_progress';
+	const STATE_COMPLETED   = 'completed';
+
+	/**
+	 * Integration instance.
+	 *
+	 * @var POSIntegration
+	 */
+	private $integration;
+
+	/**
+	 * Dependency injector.
+	 *
+	 * @param POSIntegration $integration The integration instance.
+	 */
+	public function init( POSIntegration $integration ) {
+		$this->integration = $integration;
+	}
+
+	/**
+	 * Register hooks for the async generator.
+	 *
+	 * @return void
+	 */
+	public function register_hooks(): void {
+		add_action( self::FEED_GENERATION_ACTION, [ $this, 'feed_generation_action' ] );
+		add_action( self::FEED_DELETION_ACTION, [ $this, 'feed_deletion_action' ] );
+	}
+
+	/**
+	 * Returns the current feed generation status.
+	 * Initiates one if not already running.
+	 *
+	 * @return array The feed generation status.
+	 */
+	public function get_status(): array {
+		$status = get_transient( self::TRANSIENT_KEY );
```

Comment:
`TRANSIENT_KEY` data is critical for the whole process, while `*_transient` does not guarantee that it will always return data. 

I think we should switch to use `*_option`.